pkg/formulaexec/formula_exec.go

```diff
@@ -579,16 +578,20 @@ func GetBinPath() (string, wfapi.Error) {
 // - warpforge-error-formula-invalid -- when an invalid formula is provided
 // - warpforge-error-serialization -- when serialization or deserialization of a memo fails
 func execFormula(ws *workspace.Workspace, fc wfapi.FormulaAndContext, formulaConfig wfapi.FormulaExecConfig, logger logging.Logger) (wfapi.RunRecord, wfapi.Error) {
-	formula := fc.Formula
-	context := fc.Context
 	rr := wfapi.RunRecord{}
 
-	// convert formula to node
-	nFormulaAndContext, errRaw := bindnode.Wrap(&fc, wfapi.TypeSystem.TypeByName("FormulaAndContext")).LookupByString("formula")
-	if errRaw != nil {
-		// panic! this should never fail unless the TypeSystem is broken
-		panic(fmt.Sprintf("Fatal IPLD Error: bindnode.Wrap failed for FormulaAndContext: %s", errRaw))
+	if fc.Formula.Formula == nil {
+		return rr, wfapi.ErrorFormulaInvalid("no v1 Formula in FormulaCapsule")
```

Comment:
Hm, maybe this should say something about "unrecognized version" in the message?

Might be worth a different error code too?  Less sure on that -- at the moment it won't make much practical difference.  But it seems to me there's a semiotic difference between "i tried to parse this and it violated some constraints" and "i tried to parse this and it clearly wasn't a message version that I understand".

Reply:
Hang on.  On second thought, I might be blowing smoke here.  Is this branch even going to do anything right now?  If this thing is nil... then the schema parse should've flunked this data out already.  This branch should be unreachable, until the schema actually gets more than one member in that capsule type.